javascript/modules/client/src/client.ts

```diff
@@ -282,32 +308,124 @@ export class Client extends ClientCore implements IClient {
       override.value = amount;
     }
 
-    return daoInstance.estimateGas.deposit(
-      tokenAddress,
-      amount,
-      reference,
-      override,
-    ).then((gasLimit) => {
-      return this.web3.getApproximateGasFee(gasLimit.toBigInt());
-    });
+    return daoInstance.estimateGas
+      .deposit(tokenAddress, amount, reference, override)
+      .then(gasLimit => {
+        return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+      });
+  }
+
+  private async _getDaoBalances(
+    daoIdentifier: string
+  ): Promise<TokenBalance[]> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    const tokenBalances: TokenBalance[] = tokenList.map(
+      (token: TokenBalance["token"]) => ({
+        id: `${daoIdentifier}_${token.id}`,
```

Comment:
The caller already knows the DAO address. Why creating a compound field? 
```suggestion
        address: token.address,
```

Reply:
This is the exact same pattern that exists on the graph data structure `${daoIdentifier}_${token.id}`. I can remove it if necessary because It's nonsense to have two token addresses on a single data structure!